src/shared/components/query/QueryStore.ts

```diff
@@ -1142,11 +1145,11 @@ export class QueryStore
 		}
 	}
 
-	@computed get geneIds():string[]
+	@computed get geneIds(): string[]
 	{
 		try
 		{
-			return this.oql.query.map(line => line.gene).filter(gene => gene && gene !== 'DATATYPES');
+			return this.oql.query.map(line => line.gene);

```

Comment:
The `oql` property now holds output of the `parseOQL()` function, which evaluates both merged track and datatypes statements, and returns a plain listing of the genes and filter criteria expressed by all these syntaxes.

Reply:
ah, I see

Reply:
This is partially because of the interactions between merged track groupings and DATATYPES commands outlined in the PR description—the merged tracks cannot be flattened out without (first) evaluating the DATATYPES grouping. And most tabs, apart from the Oncoprint tracks, will keep expecting a flat list without grouping, even after I present the pull request to make the Oncoprint render merged tracks (currently waiting for this one to be in `rc`).